source/common/upstream/outlier_detection_impl.h

```diff
@@ -37,6 +37,39 @@ class DetectorImplFactory {
                                       EventLoggerPtr event_logger);
 };
 
+struct SRAccumulatorBucket {
+  std::atomic<uint64_t> success_rq_counter_;
+  std::atomic<uint64_t> total_rq_counter_;
+};
+
+/**
+ * The S(uccess)R(ate)AccumulatorImpl uses the SRAccumulatorBucket to get per host Success Rate
+ * stats. This implementation has a fixed window size of time, and thus only needs a
+ * bucket to write to, and a bucket to accumulate/run stats over.
+ */
+class SRAccumulatorImpl {
+public:
+  SRAccumulatorImpl()
+      : current_sr_bucket_(new SRAccumulatorBucket()),
+        backup_sr_bucket_(new SRAccumulatorBucket()) {}
+
+  SRAccumulatorBucket* getCurrentWriter();
```

Comment:
this is poorly named, as it does the swap also. Make name more descriptive and add comment.